executor/builder.go

```diff
@@ -988,7 +988,47 @@ func (b *executorBuilder) buildDDL(v *plannercore.DDL) Executor {
 		if len(v.Statement.(*ast.AlterTableStmt).Specs) > 1 && b.Ti != nil {
 			b.Ti.UseMultiSchemaChange = true
 		}
+	case *ast.CreateTableStmt:
+		stmt := v.Statement.(*ast.CreateTableStmt)
+		if stmt != nil && stmt.Partition != nil {
+			if strings.EqualFold(b.ctx.GetSessionVars().EnableTablePartition, "OFF") {
+				break
+			}
+
+			s := stmt.Partition
+			if b.Ti.PartitionTelemetry == nil {
+				b.Ti.PartitionTelemetry = &PartitionTelemetryInfo{}
+			}
+			b.Ti.PartitionTelemetry.TablePartitionMaxPartitionsNum = mathutil.Max(s.Num, uint64(len(s.Definitions)))
+			b.Ti.PartitionTelemetry.UseTablePartition = true
+
+			switch s.Tp {
+			case model.PartitionTypeRange:
+				if s.Sub == nil {
+					if len(s.ColumnNames) > 0 {
+						b.Ti.PartitionTelemetry.UseTablePartitionRangeColumns = true
+					} else {
+						b.Ti.PartitionTelemetry.UseTablePartitionRange = true
+					}
+				}
+			case model.PartitionTypeHash:
+				if !s.Linear && s.Sub == nil {
+					b.Ti.PartitionTelemetry.UseTablePartitionHash = true
+				}
+			case model.PartitionTypeList:
+				enable := b.ctx.GetSessionVars().EnableListTablePartition
+				if s.Sub == nil && enable {
+					if len(s.ColumnNames) > 0 {
+						b.Ti.PartitionTelemetry.UseTablePartitionListColumns = true
+					} else {
+						b.Ti.PartitionTelemetry.UseTablePartitionList = true
+					}
+				}
+
+			}
+		}
 	}
+
```

Comment:
Telemetry works on the instance status, rather than on an action
If you add the telemetry here, the first second, you get the tidb using partition ... the next second, it's not using partition!

Reply:
It's better to check the infoschema table's meta information to decided whether partition is used or not.

Reply:
I see... you use prometheus metrics counter for the telemetry data.